command/agent/cache/handler.go

```diff
@@ -20,10 +21,23 @@ import (
 	"github.com/hashicorp/vault/sdk/logical"
 )
 
-func Handler(ctx context.Context, logger hclog.Logger, proxier Proxier, inmemSink sink.Sink) http.Handler {
+func Handler(ctx context.Context, logger hclog.Logger, proxier Proxier, inmemSink sink.Sink, requireRequestHeader bool) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		logger.Info("received request", "method", r.Method, "path", r.URL.Path)
 
+		// Check for the required request header
+		if requireRequestHeader {
+			val, ok := r.Header["Vault-Request"]
+			if !ok || !reflect.DeepEqual(val, []string{"true"}) {
+				logical.RespondError(w, http.StatusPreconditionFailed, errors.New("missing 'Vault-Request' header"))
+				return
+			}
+
+			// Remove the required request header
+			delete(r.Header, "Vault-Request")
```

Comment:
I wonder if it makes sense to always remove this header, even when the feature not's enabled?  Just to make sure that Vault sees the same requests coming in regardless of the setting.

Reply:
That is a good point and I do think we should change something here.  However, since the CLI now always includes the header value even on requests sent directly to a Vault Server, the Server is going to see the header if there is no Agent set up in front of it.  I wonder if we should just stop deleting the header here?

Reply:
Seems the simplest option.

Reply:
I removed that line; the header is no longer deleted.